File: gaia-sdk-python/gaia_sdk/graphql/request/type/CreateKnowledge.py

from gaia_sdk.graphql.request.type.CreatedEdgeImpulse import CreatedEdgeImpulse
from gaia_sdk.graphql.request.type.CreatedCodeImpulse import CreatedCodeImpulse
from gaia_sdk.graphql.request.type.CreatedPromptImpulse import CreatedPromptImpulse
from gaia_sdk.graphql.request.type.CreatedIntentImpulse import CreatedIntentImpulse
from gaia_sdk.graphql.request.type.CreatedSkillImpulse import CreatedSkillImpulse
from gaia_sdk.graphql.request.type.CreatedFulfilmentImpulse import CreatedFulfilmentImpulse
from gaia_sdk.graphql.request.type.CreatedUserImpulse import CreatedUserImpulse
from gaia_sdk.graphql.request.type.CreatedSkillProvisionImpulse import CreatedSkillProvisionImpulse
from gaia_sdk.graphql.request.type.CreatedTenantImpulse import CreatedTenantImpulse
from gaia_sdk.graphql.request.type.CreatedStatementImpulse import CreatedStatementImpulse
from gaia_sdk.graphql.request.type.CreatedRoleImpulse import CreatedRoleImpulse
from gaia_sdk.graphql.request.type.CreatedBehaviourImpulse import CreatedBehaviourImpulse
from gaia_sdk.graphql.request.type.CreatedIdentityImpulse import CreatedIdentityImpulse
from gaia_sdk.graphql.request.type.CreatedApiKeyImpulse import CreatedApiKeyImpulse
from gaia_sdk.graphql.request.input.CreateTenantImpulse import CreateTenantImpulse
from gaia_sdk.graphql.request.input.CreateSkillImpulse import CreateSkillImpulse
from gaia_sdk.graphql.request.input.CreateEdgeImpulse import CreateEdgeImpulse
from gaia_sdk.graphql.request.input.CreateIdentityImpulse import CreateIdentityImpulse
from gaia_sdk.graphql.request.input.CreateCodeImpulse import CreateCodeImpulse
from gaia_sdk.graphql.request.input.CreateFulfilmentImpulse import CreateFulfilmentImpulse
from gaia_sdk.graphql.request.input.CreateStatementImpulse import CreateStatementImpulse
from gaia_sdk.graphql.request.input.CreateSkillProvisionImpulse import CreateSkillProvisionImpulse
from gaia_sdk.graphql.request.input.CreateRoleImpulse import CreateRoleImpulse
from gaia_sdk.graphql.request.input.CreateIntentImpulse import CreateIntentImpulse
from gaia_sdk.graphql.request.input.CreatePromptImpulse import CreatePromptImpulse
from gaia_sdk.graphql.request.input.CreateBehaviourImpulse import CreateBehaviourImpulse
from gaia_sdk.graphql.request.input.CreateUserImpulse import CreateUserImpulse
from gaia_sdk.graphql.request.input.CreateApiKeyImpulse import CreateApiKeyImpulse

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class CreateKnowledge(list):

    """
    creates a list of identities with the given specifications
    """
    def identities(self, impulses: List[CreateIdentityImpulse], config: Callable[['CreatedIdentityImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedIdentityImpulse()
            config(entity)
            return f'identities(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of tenants with the given specifications
    """
    def tenants(self, impulses: List[CreateTenantImpulse], config: Callable[['CreatedTenantImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedTenantImpulse()
            config(entity)
            return f'tenants(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of users with the given specifications
    """
    def users(self, impulses: List[CreateUserImpulse], config: Callable[['CreatedUserImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedUserImpulse()
            config(entity)
            return f'users(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of api keys with the given specifications
    """
    def api_keys(self, impulses: List[CreateApiKeyImpulse], config: Callable[['CreatedApiKeyImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedApiKeyImpulse()
            config(entity)
            return f'apiKeys(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of roles with the given specifications
    """
    def roles(self, impulses: List[CreateRoleImpulse], config: Callable[['CreatedRoleImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedRoleImpulse()
            config(entity)
            return f'roles(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of intents with the given specifications
    """
    def intents(self, impulses: List[CreateIntentImpulse], config: Callable[['CreatedIntentImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedIntentImpulse()
            config(entity)
            return f'intents(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of prompts with the given specifications
    """
    def prompts(self, impulses: List[CreatePromptImpulse], config: Callable[['CreatedPromptImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedPromptImpulse()
            config(entity)
            return f'prompts(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of statements with the given specifications
    """
    def statements(self, impulses: List[CreateStatementImpulse], config: Callable[['CreatedStatementImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedStatementImpulse()
            config(entity)
            return f'statements(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of fulfilments with the given specifications
    """
    def fulfilments(self, impulses: List[CreateFulfilmentImpulse], config: Callable[['CreatedFulfilmentImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedFulfilmentImpulse()
            config(entity)
            return f'fulfilments(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of behaviours with the given specifications
    """
    def behaviours(self, impulses: List[CreateBehaviourImpulse], config: Callable[['CreatedBehaviourImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedBehaviourImpulse()
            config(entity)
            return f'behaviours(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of codes with the given specifications
    """
    def codes(self, impulses: List[CreateCodeImpulse], config: Callable[['CreatedCodeImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedCodeImpulse()
            config(entity)
            return f'codes(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of edges with the given specifications
    """
    def edges(self, impulses: List[CreateEdgeImpulse], config: Callable[['CreatedEdgeImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedEdgeImpulse()
            config(entity)
            return f'edges(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of skills with the given specifications
    """
    def skills(self, impulses: List[CreateSkillImpulse], config: Callable[['CreatedSkillImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedSkillImpulse()
            config(entity)
            return f'skills(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    """
    creates a list of skill provisions with the given specifications
    """
    def skill_provisions(self, impulses: List[CreateSkillProvisionImpulse], config: Callable[['CreatedSkillProvisionImpulse'], None]):
        def callback(registry: VariableRegistry):
            name1 = registry.register("impulses", impulses)
            entity = CreatedSkillProvisionImpulse()
            config(entity)
            return f'skillProvisions(impulses:{name1})' + '{' + entity.render(registry) + '}'
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/CreateKnowledge.ts

import {CreatedEdgeImpulse} from "./CreatedEdgeImpulse";
import {CreatedCodeImpulse} from "./CreatedCodeImpulse";
import {CreatedPromptImpulse} from "./CreatedPromptImpulse";
import {CreatedIntentImpulse} from "./CreatedIntentImpulse";
import {CreatedSkillImpulse} from "./CreatedSkillImpulse";
import {CreatedFulfilmentImpulse} from "./CreatedFulfilmentImpulse";
import {CreatedUserImpulse} from "./CreatedUserImpulse";
import {CreatedSkillProvisionImpulse} from "./CreatedSkillProvisionImpulse";
import {CreatedTenantImpulse} from "./CreatedTenantImpulse";
import {CreatedStatementImpulse} from "./CreatedStatementImpulse";
import {CreatedRoleImpulse} from "./CreatedRoleImpulse";
import {CreatedBehaviourImpulse} from "./CreatedBehaviourImpulse";
import {CreatedIdentityImpulse} from "./CreatedIdentityImpulse";
import {CreatedApiKeyImpulse} from "./CreatedApiKeyImpulse";
import {CreateTenantImpulse} from "../input/CreateTenantImpulse";
import {CreateSkillImpulse} from "../input/CreateSkillImpulse";
import {CreateEdgeImpulse} from "../input/CreateEdgeImpulse";
import {CreateIdentityImpulse} from "../input/CreateIdentityImpulse";
import {CreateCodeImpulse} from "../input/CreateCodeImpulse";
import {CreateFulfilmentImpulse} from "../input/CreateFulfilmentImpulse";
import {CreateStatementImpulse} from "../input/CreateStatementImpulse";
import {CreateSkillProvisionImpulse} from "../input/CreateSkillProvisionImpulse";
import {CreateRoleImpulse} from "../input/CreateRoleImpulse";
import {CreateIntentImpulse} from "../input/CreateIntentImpulse";
import {CreatePromptImpulse} from "../input/CreatePromptImpulse";
import {CreateBehaviourImpulse} from "../input/CreateBehaviourImpulse";
import {CreateUserImpulse} from "../input/CreateUserImpulse";
import {CreateApiKeyImpulse} from "../input/CreateApiKeyImpulse";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";

export class CreateKnowledge extends Array<(_:VariableRegistry) => string> {
public _typeName = "CreateKnowledge";
    /**
     * creates a list of identities with the given specifications
     */
    public identities = (impulses: [CreateIdentityImpulse]|undefined, config: (_:CreatedIdentityImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedIdentityImpulse();
        config(entity);
        return `identities(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of tenants with the given specifications
     */
    public tenants = (impulses: [CreateTenantImpulse]|undefined, config: (_:CreatedTenantImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedTenantImpulse();
        config(entity);
        return `tenants(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of users with the given specifications
     */
    public users = (impulses: [CreateUserImpulse]|undefined, config: (_:CreatedUserImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedUserImpulse();
        config(entity);
        return `users(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of api keys with the given specifications
     */
    public apiKeys = (impulses: [CreateApiKeyImpulse]|undefined, config: (_:CreatedApiKeyImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedApiKeyImpulse();
        config(entity);
        return `apiKeys(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of roles with the given specifications
     */
    public roles = (impulses: [CreateRoleImpulse]|undefined, config: (_:CreatedRoleImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedRoleImpulse();
        config(entity);
        return `roles(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of intents with the given specifications
     */
    public intents = (impulses: [CreateIntentImpulse]|undefined, config: (_:CreatedIntentImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedIntentImpulse();
        config(entity);
        return `intents(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of prompts with the given specifications
     */
    public prompts = (impulses: [CreatePromptImpulse]|undefined, config: (_:CreatedPromptImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedPromptImpulse();
        config(entity);
        return `prompts(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of statements with the given specifications
     */
    public statements = (impulses: [CreateStatementImpulse]|undefined, config: (_:CreatedStatementImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedStatementImpulse();
        config(entity);
        return `statements(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of fulfilments with the given specifications
     */
    public fulfilments = (impulses: [CreateFulfilmentImpulse]|undefined, config: (_:CreatedFulfilmentImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedFulfilmentImpulse();
        config(entity);
        return `fulfilments(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of behaviours with the given specifications
     */
    public behaviours = (impulses: [CreateBehaviourImpulse]|undefined, config: (_:CreatedBehaviourImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedBehaviourImpulse();
        config(entity);
        return `behaviours(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of codes with the given specifications
     */
    public codes = (impulses: [CreateCodeImpulse]|undefined, config: (_:CreatedCodeImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedCodeImpulse();
        config(entity);
        return `codes(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of edges with the given specifications
     */
    public edges = (impulses: [CreateEdgeImpulse]|undefined, config: (_:CreatedEdgeImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedEdgeImpulse();
        config(entity);
        return `edges(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of skills with the given specifications
     */
    public skills = (impulses: [CreateSkillImpulse]|undefined, config: (_:CreatedSkillImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedSkillImpulse();
        config(entity);
        return `skills(impulses:${name1}){` + entity.render(registry) + "}"
    });

    /**
     * creates a list of skill provisions with the given specifications
     */
    public skillProvisions = (impulses: [CreateSkillProvisionImpulse]|undefined, config: (_:CreatedSkillProvisionImpulse) => void) => this.push((registry) => {
        const name1 = registry.register("impulses", impulses);
        const entity = new CreatedSkillProvisionImpulse();
        config(entity);
        return `skillProvisions(impulses:${name1}){` + entity.render(registry) + "}"
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

